File: March/CPP/Find All Duplicates In an Array/Find All Duplicates In an array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) 
    {
       vector<int> array;  

       int N = nums.size();

       for ( int i = 0 ; i < N ; i++ )
       {
        if ( nums[abs(nums[i])-1] > 0 )
        {
            nums[abs(nums[i])-1] = nums[abs(nums[i])-1]*(-1);
        }

        else
        {
            array.push_back(abs(nums[i]));
        }
       } 
       return array; 
    }
};

File: March/Python/Find All Duplicates in an Array/Find All Duplicates in an Array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        output = []

        for n in nums:
            m = abs(n)
            if nums[m-1] < 0:
                output.append(m)
            else:
                nums[m-1] *= -1

        return output
